static/js/vue-cdr-access/tests/unit/abstract.spec.js

```diff
@@ -0,0 +1,182 @@
+
+    it('show full text for short abstract', () => {
+        expect(wrapper.vm.truncateAbstract).toBe(false);
+        expect(wrapper.find('.abstract-text').exists()).toBe(false);
+        expect(wrapper.find('.abstract').text()).toEqual(recordData.briefObject.abstractText);
+    });
+
+    it('truncates long abstracts', async () => {
+        expect(wrapper.vm.truncateAbstract).toBe(false);
+        let longAbstract = cloneDeep(recordData);
+        longAbstract.briefObject.abstractText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,' +
+            ' sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Id cursus metus aliquam eleifend' +
+            ' mi in nulla. In hendrerit gravida rutrum quisque non. Libero volutpat sed cras ornare arcu dui. Diam' +
+            ' maecenas sed enim ut sem viverra aliquet eget. Malesuada fames ac turpis egestas sed tempus. In' +
+            ' egestas erat imperdiet sed euismod nisi. Ut pharetra sit amet aliquam id diam maecenas ultricies.' +
+            ' Eget nullam non nisi est. Risus viverra adipiscing at in tellus integer feugiat scelerisque varius.' +
+            ' Sit amet risus nullam eget felis eget nunc lobortis. Dui vivamus arcu felis bibendum ut tristique' +
+            ' et egestas quis. Eget nunc scelerisque viverra mauris in aliquam sem. Facilisi nullam vehicula' +
+            ' ipsum a. Odio facilisis mauris sit amet massa vitae tortor. Donec et odio pellentesque diam.' +
+            ' Commodo quis imperdiet massa tincidunt. Sagittis eu volutpat odio facilisis mauris sit amet massa.' +
+            'Nunc aliquet bibendum enim facilisis gravida neque convallis. Arcu non sodales neque sodales' +
+            ' ut etiam. Posuere lorem ipsum dolor sit amet consectetur. Non pulvinar neque laoreet suspendisse' +
+            ' interdum consectetur libero id faucibus. Netus et malesuada fames ac turpis. Sit amet mauris' +
+            ' commodo quis imperdiet massa tincidunt. Venenatis urna cursus eget nunc. Aliquet lectus proin' +
+            ' nibh nisl condimentum id. Pellentesque habitant morbi tristique senectus et netus et malesuada.' +
+            ' Pulvinar elementum integer enim neque volutpat ac tincidunt. Sed viverra tellus in hac habitasse' +
+            ' platea. Dui id ornare arcu odio ut sem. Tincidunt arcu non sodales neque sodales ut etiam sit.' +
+            ' Quisque non tellus orci ac auctor augue. Vitae aliquet nec ullamcorper sit amet risus. Tempor' +
+            ' nec feugiat nisl pretium. Posuere lorem ipsum dolor sit amet consectetur adipiscing elit.' +
+            ' Pellentesque eu tincidunt tortor aliquam.';
+
+        await wrapper.setProps({
+            recordData: longAbstract
+        });
+
+        let show_more = wrapper.find('.abstract-text');
+        expect(wrapper.vm.truncateAbstract).toBe(true);
+        expect(show_more.exists()).toBe(true);
+        expect(wrapper.vm.abstractLinkText).toEqual('Read more');
+    });
```

Comment:
It would be good to also test that clicking the "Read more" link expands the abstract, and clicking the "Read less" button hides the extra.